### PR TITLE
Movies

### PR DESCRIPTION
**Pull request by divinecharlotte**
## IN THIS PR I IMPLEMENTED THE FOLLOWING:
- Created `Movie` class in a separate .rb file.
- Created `Source` class with an association to the Item class (in a separate .rb file).
- All Movie class properties visible in the diagram are defined and set up in the constructor method.
- All Source class properties visible in the diagram are defined and set up in the constructor method.

**Implemented  methods:**

- [ ] add_item method in the Source class

1. it takes an instance of the Item class as an input
2. it adds the input item to the collection of items
3. it adds self as a property of the item object (by using the correct setter from the item object)

- [ ] can_be_archived?() in the Movie class

1. it overrides the method from the parent class
2. it returns true if parent's method returns true OR if silent equals true
3. otherwise, it should return false

### CODE REVIEW

File: movie.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    super(publish_date)
+    @id = Random.rand(1...1000)
+    @name = name
+    @silet = silet
+  end
+
+  def can_be_archived?
+    super && @silet == true
+  end
+end
```
Comment (Sanja969) [inline movie.rb]: The project requirement for this method is to return true if parent's method returns true OR if silent equals true, so you should put || logical operator instrad of &&.

**Review by Sanja969 — CHANGES_REQUESTED** [review]
Hi @divinecharlotte 

Good job so far! I see you wrote clean code. However, there are some issues that you still that should be changed

**Highlights**

> No linter errors
> Descriptive PR

**Required Changes recycle** :recycle: 
Check the comments under the review.

**Optional suggestions**
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.


Cheers and Happy coding! clapclapclap

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

File: movie.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require_relative 'item'
+class Movie < Item
+  attr_accessor :id, :name, :publish_date, :silet
+
```
Comment (Sanja969) [inline movie.rb]: Use attr_reader for id, I shouldn't be changed from created object

Comment (divinecharlotte) [reply movie.rb]: thanks, @Sanja969 am adding on the required changes as soon as possible

**Review by Sanja969 — APPROVED** [review]
Hi @divinecharlotte 

## Highlights!
- No linter errors ✔️ 
- Descriptive pull request ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

**Pull request merged**